packages/fury-adapter-oas3-parser/lib/parser/parseArray.js

```diff
@@ -0,0 +1,96 @@
+const R = require('ramda');
+const {
+  isAnnotation, isNotAnnotation, isParseResult, isArray,
+} = require('../predicates');
+const { createWarning } = require('./annotations');
+const pipeParseResult = require('../pipeParseResult');
+
+const parseResultHasErrors = parseResult => !parseResult.errors.isEmpty;
+
+/**
+ * Transform every non-annotation element in the parse result and then flatten all of the results into a parse result
+ * @param transform {function}
+ * @param parseResult {ParseResult}
+ */
+const chainParseResult = R.curry((transform, parseResult) => {
+  const result = R.chain(transform, parseResult);
+
+  if (!result.errors.isEmpty) {
+    return new parseResult.constructor(result.errors);
+  }
+
+  return result;
+});
+
+/**
+ * A callback for transforming a member element
+ *
+ * @callback transformMember
+ * @param member {MemberElement}
+ * @returns {Element} Either a ParseResult to be unwrapped, or an element
+ */
+
+/**
+ * Function for parsing each value in an array using the given transform callback.
+ *
+ * The given transform callback can return a parse result containing
+ * annotations to be merged into the final parse result.
+ *
+ * When the transform callback returns an error, parseArray will return a
+ * ParseResult only containing errors.
+ *
+ * If the transform callback returns a parse result without any non-annotation
+ * elements then the member will be removed from the resultant object return in the parse result.
+ *
+ * Splits up each value from an array, invokes the given parseValue
+ * transformation and then reconstructs a parse result containing all
+ * of the results.
+ *
+ * |------->------->------------------->--------------------|
+ * |       > value > parseValue(value) >                    |
+ * | Array > value > parseValue(value) > ParseResult<Array> |
+ * |       > value > parseValue(value) >                    |
+ * |------->------->------------------->--------------------|
+ *
+ * @param namespace
```

Comment:
The signature of the function is `context` not namespace.

```suggestion
 * @param context
```